coarse/api/typogrammar.py

```
@@ -3,13 +3,20 @@
+    
+    Accepts a PaperObj payload, performs typo and grammar checking, and returns the result. Raises an HTTP 500 error if processing fails.
+    """
     try:
-        result = await check_typo_grammar(paper_obj)
-        return result
+        result = await check_typo_grammar(paper_obj.root)
+        return result  # no .dict() here
     except Exception as e:
-        raise HTTPException(status_code=500, detail=f"An error occurred while processing the PDF: {str(e)}")
\ No newline at end of file
+        raise HTTPException(status_code=500, detail=f"An error occurred while processing the PDF: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)